logstash-core/spec/logstash/instrument/wrapped_write_client_spec.rb

```diff
@@ -18,39 +18,54 @@
 
   subject { described_class.new(write_client, pipeline, metric, plugin) }
 
+  def threaded_read_client(read_client)
```

Comment:
That will be a bit harder, since we have to *join* on the thread to go to the next spec.
Moving the expectation out, will mean having to pass a variable as a reference to the thread and might cause thread safety issue.